File: spec/java/src/io/kaitai/struct/spec/TestSwitchRepeatExpr.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.SwitchRepeatExpr;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestSwitchRepeatExpr extends CommonSpec {
    @Test
    public void testSwitchRepeatExpr() throws Exception {
        SwitchRepeatExpr r = SwitchRepeatExpr.fromFile(SRC_DIR + "switch_tlv.bin");

        assertEquals(r.code(), 0x11);
        assertEquals(r.size(), 9);
        assertEquals(
                ((SwitchRepeatExpr.One) r.body().get(0)).first(),
                "Stuff\0Me\0".getBytes()
        );
    }
}

File: spec/cpp_stl_98/test_switch_repeat_expr.cpp
#include <boost/test/unit_test.hpp>

#include <switch_repeat_expr.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_switch_repeat_expr) {
    std::ifstream ifs("src/switch_tlv.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    switch_repeat_expr_t* r = new switch_repeat_expr_t(&ks);

    BOOST_CHECK_EQUAL(r->code(), 0x11);
    BOOST_CHECK_EQUAL(r->size(), 9);
    BOOST_CHECK_EQUAL((static_cast<switch_repeat_expr_t::one_t*>(r->body()->at(0)))->first(), std::string("Stuff\0Me\0", 9));

    delete r;
}
